models/avt/optimism/avt_optimism_v1_perpetual_trades.sql

```diff
@@ -0,0 +1,154 @@
+	materialized = 'incremental',
+	file_format = 'delta',
+	incremental_strategy = 'merge',
+	unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["optimism"]\',
+                                "project",
+                                "avt",
+                                \'["kaiblade"]\') }}'
+	)
+}}
+
+{%
+set project_start_date = '2023-03-27' %}
+
+WITH all_executed_positions AS (
+SELECT block_time,
+       block_number,
+       index,
+       contract_address,
+       tx_hash,
+       bytearray_substring(data, 45, 20)                       AS account,
+       bytearray_substring(data, 77, 20)                       AS collateralToken,
+       bytearray_substring(data, 109, 20)                      AS indexToken,
+       varbinary_to_int256(bytearray_substring(data, 141, 20)) AS collateralDelta,
+       varbinary_to_int256(bytearray_substring(data, 173, 20)) AS sizeDelta,
+       (CASE
+            WHEN varbinary_to_int256(bytearray_substring(data, 205, 20)) = 1 THEN true
+            ELSE false
+           END)                                                AS isLong,
+       varbinary_to_int256(bytearray_substring(data, 237, 20)) AS price,
+       'Open'                                                  AS trade_type
+FROM {{ source('optimism', 'logs') }}
+
+WHERE contract_address = 0x24ee37267842a525c66fe37cd0da749150e89866
+  AND topic0 = 0x2fe68525253654c21998f35787a8d0f361905ef647c854092430ab65f2f15022
+  AND tx_hash IN ( SELECT evt_tx_hash FROM {{ source('avt_optimism', 'PositionRouter_evt_ExecuteIncreasePosition') }} )
+    {% if not is_incremental() %}
+  AND block_time >= DATE '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+  AND block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
+    {% endif %}
+
+UNION
+
+SELECT 
+    block_time,
+    block_number,
+    index,
+    contract_address,
+    tx_hash,
+    bytearray_substring(data,45,20) AS account,
+    bytearray_substring(data,77,20) AS collateralToken,
+    bytearray_substring(data,109,20) AS indexToken,
+    varbinary_to_int256(bytearray_substring(data,141,20)) AS collateralDelta,
+    varbinary_to_int256(bytearray_substring(data,173,20)) AS sizeDelta,
+    (CASE
+    WHEN varbinary_to_int256(bytearray_substring(data,205,20)) = 1 THEN true
+    ELSE false
+    END) AS isLong,
+    varbinary_to_int256(bytearray_substring(data,237,20)) AS price,
+    'Close' AS trade_type
+FROM {{ source('optimism','logs') }}
+
+WHERE contract_address = 0x24ee37267842a525c66fe37cd0da749150e89866
+  AND topic0 = 0x93d75d64d1f84fc6f430a64fc578bdd4c1e090e90ea2d51773e626d19de56d30
+  AND tx_hash IN ( SELECT evt_tx_hash FROM {{ source('avt_optimism', 'PositionRouter_evt_ExecuteDecreasePosition') }} )
+    {% if not is_incremental() %}
+  AND block_time >= DATE '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+  AND block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
```

Comment:
You can use `incremental_predicate` in future:
```suggestion
  AND {{ incremental_predicate('block_time') }}
```